engine/sync_manager.go

```diff
@@ -457,10 +458,11 @@ func (m *syncManager) Update(exchangeName string, p currency.Pair, a asset.Item,
 						createdCounter)
 					m.initSyncWG.Done()
 				}
+				return nil
 			}
 		}
 	}
-	return nil
+	return fmt.Errorf("%w for %s %s %s", errCouldNotSyncNewData, exchangeName, p, a)
```

Comment:
I'm not going to ask you to add full coverage of this function, but could you add a test to hit this new error at least?